packages/ui/src/component/UITransform.ts

```diff
@@ -35,15 +65,348 @@ export class UITransform extends Transform {
+          case HorizontalAlignmentFlags.Right:
+            position.x = parentRect.x + parentRect.width - rect.x - rect.width - this._right;
+            break;
+          default:
+            break;
+        }
+        switch (this.verticalAlignment) {
+          case VerticalAlignmentFlags.Top:
+            position.y = parentRect.y + parentRect.height - rect.y - rect.height - this._top;
+            break;
+          case VerticalAlignmentFlags.Middle:
+            position.y = parentRect.y + parentRect.height * 0.5 - rect.y - rect.height * 0.5 + this._middle;
+            break;
+          case VerticalAlignmentFlags.Bottom:
+          case VerticalAlignmentFlags.TopAndBottom:
+            position.y = parentRect.y - rect.y + this._bottom;
+            break;
+          default:
+            break;
+        }
+        // @ts-ignore
+        position._onValueChanged = this._onPositionChanged;
+      }
+      this._setDirtyFlagFalse(UITransformModifyFlags.LocalPosition);
+    }
+    return this._position;
+  }
+
+  override set position(value: Vector3) {
+    if (this._position !== value) {
+      this._position.copyFrom(value);
+    }
+  }
+
+  /**
+   * Local matrix.
+   * @remarks Need to re-assign after modification to ensure that the modification takes effect.
+   */
+  override get localMatrix(): Matrix {
+    if (this._isContainDirtyFlag(UITransformModifyFlags.LocalMatrix)) {
+      Matrix.affineTransformation(this._scale, this.rotationQuaternion, this.position, this._localMatrix);
+      this._setDirtyFlagFalse(UITransformModifyFlags.LocalMatrix);
+    }
+    return this._localMatrix;
+  }
+
+  override set localMatrix(value: Matrix) {
+    if (this._localMatrix !== value) {
+      this._localMatrix.copyFrom(value);
+    }
+    const { _position: position, _rotationQuaternion: rotationQuaternion, _scale: scale } = this;
+    // @ts-ignore
+    position._onValueChanged = rotationQuaternion._onValueChanged = scale._onValueChanged = null;
+    value.decompose(position, rotationQuaternion, scale);
+    // @ts-ignore
+    position._onValueChanged = this._onPositionChanged;
+    // @ts-ignore
+    rotationQuaternion._onValueChanged = this._onRotationQuaternionChanged;
+    // @ts-ignore
+    scale._onValueChanged = this._onScaleChanged;
+    this._setDirtyFlagTrue(UITransformModifyFlags.LocalEuler);
+    this._setDirtyFlagFalse(UITransformModifyFlags.LocalQuat);
+    if (!!this._horizontalAlignment || !!this._verticalAlignment) {
+      this._setDirtyFlagTrue(UITransformModifyFlags.LmLp);
+    } else {
+      this._setDirtyFlagFalse(UITransformModifyFlags.LmLp);
+    }
+    const localUniformScaling = scale.x === scale.y && scale.y === scale.z;
+    if (this._localUniformScaling !== localUniformScaling) {
+      this._localUniformScaling = localUniformScaling;
+      this._updateAllWorldFlag(UITransformModifyFlags.WmWpWeWqWsWus);
+    } else {
+      this._updateAllWorldFlag(UITransformModifyFlags.WmWpWeWqWs);
+    }
+  }
```

Comment:
If there are too many similar codes with the parent class, you should focus on maintainability. Maybe you only need to reload some sub-logic